pkg/envoy/agent.go

```diff
@@ -219,10 +220,9 @@ func (a *agent) Run(ctx context.Context) {
 				if status.err == errOutOfMemory {
 					log.Warnf("Envoy may have been out of memory killed. Check memory usage and limits.")
 				}
-				// NOTE: due to Envoy hot restart race conditions, an error from the
-				// process requires aggressive non-graceful restarts by killing all
-				// existing proxy instances
-				a.abortAll()
+				// If Envoy is killed, we will kill the whole pilot-agent process. This allows Kubernetes to handle
+				// the restart and we don't need to worry about
+				os.Exit(1)
```

Comment:
Are we sure everything else is cleaned up? Cleaning up doesn't matter on Kube but does in VMs.

Reply:
Don't pilot-agent already exit after retries have been exhausted? I guess that was an issue as well?